core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5760ResumeFeatureTest.java

```diff
@@ -175,6 +179,112 @@ public void testShouldNotCrashWithoutProject() throws Exception
         }
     }
 
+    public void testFailureWithParallelBuild() throws Exception
+    {
+        Verifier verifier = newVerifier( fourModulesTestDir.getAbsolutePath() );
+        verifier.addCliOption( "-T2" );
+        verifier.addCliOption( "-Dmodule-a.delay=1000" );
+        verifier.addCliOption( "-Dmodule-a.fail=true" );
+        verifier.addCliOption( "-Dmodule-c.fail=true" );
+
+        verifier.executeGoal( "org.apache.maven.plugins:maven-clean-plugin:clean" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-a", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-b", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-c", "1.0", "jar" );
+        verifier.deleteArtifact( "org.apache.maven.its.mng5760", "module-d", "1.0", "jar" );
+
+        try
+        {
+            verifier.executeGoal( "install" );
+            fail( "Expected this invocation to fail" );
+        }
+        catch ( final VerificationException ve )
+        {
+            // Expected to fail.
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-a", "1.0", "jar" );
+            verifier.assertArtifactPresent( "org.apache.maven.its.mng5760", "module-b", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-c", "1.0", "jar" );
+            verifier.assertArtifactNotPresent( "org.apache.maven.its.mng5760", "module-d", "1.0", "jar" );
+        }
+        finally
+        {
+            verifier.resetStreams();
+        }
+
+        // Let module-b and module-c fail, if they would have been built...
```

Comment:
You are only letting module-b fail. I think the comment is not correct, right? (if so, please also correct the one below)